src/cargo/core/package.rs

```diff
@@ -89,6 +89,17 @@ impl Package {
     pub fn generate_metadata(&self) -> Metadata {
         self.package_id().generate_metadata()
     }
+
+    pub fn find_closest_target(&self, target: &str, kind: TargetKind) -> Option<&Target> {
+        let targets = self.targets();
+
+        let mut matches = targets.iter().filter(|t| *t.kind() == kind)
+                                        .map(|t| (lev_distance(target, t.name()), t))
+                                        .filter(|&(d, _)| d < 4)
+                                        .collect::<Vec<_>>();
+        matches.sort_by(|a, b| a.0.cmp(&b.0));
```

Comment:
Can probably use [max_by_key](http://doc.rust-lang.org/std/iter/trait.Iterator.html#method.max_by_key) here instead of collect & sort?